### PR TITLE
build: Shrink binaries by 20%

### PR DESCRIPTION
**Pull request by alexec**
Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this is a chore.
* [x] The title of the PR is (a) [conventional](https://www.conventionalcommits.org/en/v1.0.0/), (b) states what changed, and (c) suffixes the related issues number. E.g. `"fix(controller): Updates such and such. Fixes #1234"`.  
* [x] I've signed the CLA.
* [x] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [x] My builds are green. Try syncing with master if they are not. 
* [x] My organization is added to USERS.md.
* [x] Sending SIGUSR1/2 should produce stack trace. Check this still works.

### CODE REVIEW

**Comment by alexec** [comment]
@jessesuen thought on removing debugging from the binaries...

**Comment by alexec** [comment]
```
kill -SIGUSR1 ${pid}
...
goroutine 1 [chan receive]:
github.com/argoproj/argo/server/apiserver.(*argoServer).Run(0xc00051e870, 0x32315c0, 0xc0006a9580, 0xaba, 0x2eea8a0)
	/Users/acollins8/go/src/github.com/argoproj/argo/server/apiserver/argoserver.go:181 +0x9d0
github.com/argoproj/argo/cmd/argo/commands.NewServerCommand.func1(0xc0004ee000, 0x41a7ba0, 0x0, 0x0, 0x0, 0x0)
	/Users/acollins8/go/src/github.com/argoproj/argo/cmd/argo/commands/server.go:111 +0x879
github.com/spf13/cobra.(*Command).execute(0xc0004ee000, 0x41a7ba0, 0x0, 0x0, 0xc0004ee000, 0x41a7ba0)
	/Users/acollins8/go/pkg/mod/github.com/spf13/cobra@v0.0.4-0.20181021141114-fe5e611709b0/command.go:762 +0x460
github.com/spf13/cobra.(*Command).ExecuteC(0xc0004ec500, 0xc000000180, 0xc000689f50, 0x100779f)
	/Users/acollins8/go/pkg/mod/github.com/spf13/cobra@v0.0.4-0.20181021141114-fe5e611709b0/command.go:852 +0x2ea
github.com/spf13/cobra.(*Command).Execute(...)
	/Users/acollins8/go/pkg/mod/github.com/spf13/cobra@v0.0.4-0.20181021141114-fe5e611709b0/command.go:800
main.main()
	/Users/acollins8/go/src/github.com/argoproj/argo/cmd/argo/main.go:17 +0x2b
```

File: Makefile [hunk 1]
```diff
@@ -455,8 +455,10 @@ test-cli: test-images cli
 
 .PHONY: clean
 clean:
-	# Delete pre-go 1.3 vendor
-	rm -Rf vendor
+ifneq ($(findstring release,$(GIT_BRANCH)),)
+	docker rm -f `docker ps -aq`
+	docker system prune -fa
+endif
```
Comment (simster7) [inline Makefile:461]: We should be careful with this. Perhaps add a confirmation message? We should avoid something like https://github.com/argoproj/argo/pull/1995

**Comment by alexec** [comment]
I'm going to abandon this PR. I don't think the benefits out-weight the risks.

**Pull request closed without merging**